README.md

```diff
@@ -579,6 +580,314 @@ Note that the subflows are an array of repeating entries - one for each iteratio
+  programs:
+    SNAP: APPLICANT_PROGRAMS_SNAP
+    TANF: APPLICANT_PROGRAMS_TANF
+    CCAP: APPLICANT_PROGRAMS_CCAP
+  incomeTypes:
+    incomeJob: APPLICANT_HAS_JOB_INCOME
+    incomeSelf: APPLICANT_HAS_SELF_EMPLOYMENT_INCOME
+    incomeUnemployment: APPLICANT_HAS_UNEMPLOYMENT_INCOME
+    incomeSocialSecurity: APPLICANT_HAS_SOCIAL_SECURITY_INCOME
+```
+
+Note that for the multi-value checkbox field mappings, `programs` and `incomeTypes` represent the
+names
+of checkbox inputs in the application, and `SNAP`, `TANF`, `CCAP`, `incomeJob`, `incomeSelf`, etc
+are
+the actual values of checkboxes within those inputs. `APPLICANT_PROGRAMS_SNAP`
+and `APPLICANT_HAS_JOB_INCOME`
+are the names of the corresponding PDF checkbox fields.
+
+#### dbFields
+
+Key value pairs for database column fields from your application. The key is the name of a database
+column and the value is the name of a field in your PDF template. Currently supported fields are:
+
+- submittedAt
+- submissionId
+- createdAt
+- updatedAt
+- flow
+
+For example:
+
+```yaml
+dbFields:
+  submittedAt: SUBMISSION_TIME
+  submissionId: SUBMISSION_ID
+  flow: FLOW_NAME
+```
+
+### Subflow specific PDF field mappings
+
+#### subflowInfo
+
+The top level key for all your subflows. `subflowInfo` holds all data related to mapping all of your
+subflows.
+
+#### totalIterations
+
+The total number of iterations possible given your PDF's template. For example, if your PDF has room
+for 5 household members, your `totalIterations` for the `household` subflow would be 5.
+
+For example:
+
+```yaml
+subflowInfo:
+  household:
+    totalIterations: 5
+```
+
+#### fields
+
+Similar to `inputFields`, `fields` is a map of input fields from your application to PDF fields in
+your
+PDF template file. Like `inputFields`, the key should always be the actual name of an input in your
+subflow, and the value can be one of two things.
+
+For single value fields like text fields and radios,
+the value will be the name of a field in your PDF template. For example:
+
+```yaml
+    fields:
+      householdMemberFirstName: LEGAL_NAME_FIRST_MEMBER
+      householdMemberLastName: LEGAL_NAME_LAST_MEMBER
+```
+
+For multi-value fields like checkboxes, the value will be a map of key value pairs where the top
+level
+key is the name of a checkbox input in your application and each internal key value pair is a map of
+checkbox field value to PDF field name. For example:
+
+```yaml
+fields:
+  incomeTypes:
+    incomeJob: INCOME_HAS_JOB_MEMBER
+    incomeSelf: INCOME_HAS_SELF_EMPLOYMENT_MEMBER
+    incomeUnemployment: INCOME_HAS_UNEMPLOYMENT_MEMBER
+```
+
+Where `incomeTypes` is the name of a checkbox input in your application,
+and `incomeJob`, `incomeSelf`,
+and `incomeUnemployment` are the actual values of checkboxes within that
+input. `INCOME_HAS_JOB_MEMBER`,
+`INCOME_HAS_SELF_EMPLOYMENT_MEMBER`, and `INCOME_HAS_UNEMPLOYMENT_MEMBER` are the names of the
+corresponding PDF checkbox fields.
+
+### Custom preparers
```

Comment:
I think it's worthwhile and necessary to talk about our default prepares so folks know what they get for "free" with our work. It will also help explain what a custom preparer does.  `The custom prepare will allow for overriding of the default preparers work as well as allows for injecting new fields, etc...`

In other words, understanding what we do for them to start with will help explain why custom preparers are useful.  They then will know that they don't have to do any default preparing, just any custom work they want.

Reply:
I know you touch on this a little, but I think it's worth describing them more before you talk about the custom ones.